[Serverless] Manual rule run docs (#5589)
elastic/security-docs: docs/serverless/alerts/alert-schema.mdx

      
    </DocCell>
  </DocRow>
    <DocRow>
    <DocCell> `kibana.alert.intended_timestamp`</DocCell>
    <DocCell>

>       Represents the estimated time when the alert was created by a manual rule run.

Reviewer: This timestamp represents the time when the alert would have been created based on the source event timestamp.

Author: @approksiu how's this:

```suggestion
      Represents the time of when the alert would have been created based on the source event timestamp. This field is only populated for alerts generated by manual rule runs.
```

Author: Added to ESS docs via https://github.com/elastic/security-docs/pull/5631/files#r1773602370.

>       Represents the time when the alert would have been created based on the source event timestamp. This field is only populated for alerts generated by manual rule runs.

Reviewer 2: With limited context in this table, "the alert _would have been_ created" is unclear. Sounds like an alert _almost_ got created, but wasn't?

Author: @nkhristinin @approksiu can you help out here? Is the following revision still accurate: 

```suggestion
      Represents the approximate time of the source event that prompted the alert. This field is only populated for alerts generated by manual rule runs.
```

Reviewer: Looks good, just not sure about "approximate".

Author: Would "estimated" be more appropriate for the description?

Reviewer 3: Hey -> This field is only populated for alerts generated by manual rule runs.

This is not correct, as it populated for all alerts. It's useful for manual rule runs, to be able understand when alert "should" be generated if there were no gaps.

For regular alerts value in this field should be the same as `kibana.alert.rule.execution.timestamp`

> Represents the approximate time of the source event that prompted the alert.

Not sure if it's super correct, but maybe it fine for docs.

Just to give you example:

Let's say current time: 4 Oct, 12:00
Source event time 3 Oct, 13:00

And we manually run rule with 6 hours lookback at 3 Oct, 16:30-17:00

intentioned timestamp will have value 3 Oct, 17:00

Author: @nkhristinin here's what I've gathered from your response -- let me know whether my understanding is correct:

- The `kibana.alert.intended_timestamp` field is populated in all alert documents, regardless of whether the alert was created by a scheduled run or a manual run.
- For alerts created by scheduled runs, the `kibana.alert.intended_timestamp` value will be identical to the value in the `kibana.alert.rule.execution.timestamp` field. 
  - I couldn't find a definition for the `kibana.alert.rule.execution.timestamp` field in the Kibana or Security docs, and it's not defined in the UI, so I'm not sure what this field shows. My best guess is that shows when the rule was triggered and is usually close to the alert's `@timestamp`.
- Based on your example, alerts created by manual runs will display a value that's identical to the upper threshold of the range specified for the manual run. 
   - If this ^ is true, how do users benefit from knowing this information and how should they use it?

Reviewer 3: >The kibana.alert.intended_timestamp field is populated in all alert documents, regardless of whether the alert was created by a scheduled run or a manual run.

yes

> `kibana.alert.rule.execution.timestamp` - from the code perspective it's actual time when we try to ingest create alerts and try to ingest it.

> usually close to the alert's @timestamp. - yes

> will display a value that's identical to the upper threshold of the range specified for the manual run. - 

In my example it's correct, but if the manual rule run range is 14:00-20:00, the intended timestamp still has a value 3 Oct, 17:00, because 1 manual rule run contains several rule executions.

>how do users benefit from knowing this information and how should they use it?

The main idea, is to compare intended_timestamp with other timestamps and check that this alert is from manual rule run, and the time when it can be generated as if it was regular execution

<img width="582" alt="Screenshot 2024-10-07 at 09 50 11" src="https://github.com/user-attachments/assets/4647147f-c824-41de-b900-f43c79ceaf7d">

Reviewer 2: Could we just use a really simple description like this?
```suggestion
      Represents the source event timestamp that generated the alert.
```

Or, if it's necessary to explain how this timestamp is different for manual rule runs (to invite the user to compare them):

```suggestion
      Represents the source event timestamp that generated the alert. For manual rule runs, this value might differ from other alert timestamps such as `@timestamp` (when the alert was generated) or `kibana.alert.rule.execution.timestamp` (when the rule was run).
```

Author: Sorry, posting this late. There was some additional discussion about this field on Slack, and here's what we settled on:

> Shows the alert’s estimated timestamp, had the alert been created when the source event initially occurred. The value in this field is determined by the way the rule was run:
> 
> - **Scheduled run**: Alerts created by scheduled runs have the same timestamp as the `kibana.alert.rule.execution.timestamp` field, which shows when the rule was executed. 
> - **Manual run**: Alerts created by manual runs have a timestamp that falls within the time range specified for the manual run. For example, if you set a rule to manually run from `10/01/2024 05:00 PM` to `10/07/2024 05:00 PM`, the `kibana.alert.intended_timestamp` value will be a date and time within that range. 

Let me know if it's still confusing or unclear @joepeeples.

Reviewer 2: That all sounds good, how about this slight tweak for further clarity:

> For example, if you set a rule to manually run on event data from `10/01/2024 05:00 PM` to `10/07/2024 05:00 PM`

At first I thought this was saying the rule would be actively running from 10/01 to 10/07, but I think it means that it's searching event data collected from that time period, right?

Author: Added in [1f6f6d5](https://github.com/elastic/security-docs/pull/5589/commits/1f6f6d5ebaf29e9589973034c63df3825858ec7b).
